pkg/reference/ref.go

```diff
@@ -1,11 +1,32 @@
 package reference
 
 import (
-	kapi "k8s.io/api/core/v1"
 	"reflect"
+	"regexp"
 	"strings"
+
+	kapi "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+func ObjectKey(obj metav1.Object) client.ObjectKey {
```

Comment:
I don't understand why this is part of the `ref` package.  The usages don't seem like a valuable change.  It's effectively changing 1 line of code with another.

Reply:
Looks like this converts a kubernetes resource (with more parameters than just a namespaced name) into an object key. It's only used in 2 places, though, so leaving it inline would be fine. Danil won't be working on this PR anymore, though, so the easiest thing to do with this sort of thing would be a follow-on PR with whatever changes are desired from this one.

Reply:
Do you think the undesirable changes could be enumerated in an issue?  I'm concerned they'll fall through the cracks.

Reply:
Sure. That makes sense.